jwt/__init__.py

```diff
@@ -128,10 +129,11 @@ def verify_jwt(jwt,
     if alg not in allowed_algs:
         raise _JWTError('algorithm not allowed: ' + alg)
 
-    claims = jws.utils.from_base64(claims).decode()
+    claims = jws.utils.from_base64(claims).decode(encoding)
 
     if pub_key:
-        jws.verify(str(header), str(claims), str(sig), pub_key, True)
+        jws.verify(header.encode(encoding), claims.encode(encoding),
+                   sig.encode(encoding), pub_key, True)
```

Comment:
`jws.verify` calls `json.loads` on `header` and `claims` so we need to encode them UTF-8 (they will be Unicode strings because we decoded them).
`jws.verify` ends up calling `urlsafe_b64decode` for `sig`, but before that it encodes `sig` as UTF-8 if it's unicode.
`sig` is going to be Unicode on Python 3 and a byte string on Python 2. Now it shouldn't matter what we do because the signature will be base64 encoded anyway but I don't think we need to encode it here. We should be able to pass it through without doing anything to it.